### ISSUE TITLE
Getting Terraform version 1.0.6 is not supported

### ISSUE THREAD

**Issue by suchitgupta01** [terraform-google-modules/terraform-google-data-fusion#53]
### TL;DR

My resource 
`module "data_fusion" {
  source  = "terraform-google-modules/data-fusion/google"
  version = "1.0.0"

  name    = "df-test-instance"
  project = "df-test-instance"
  region  = "us-central1"
}`

And it keeps failing with error:

`Step #1 - "tf plan validate all": - vault_iam in ../../modules/vault_iam
Step #1 - "tf plan validate all": ╷
Step #1 - "tf plan validate all": │ Error: Unsupported Terraform Core version
Step #1 - "tf plan validate all": │ 
Step #1 - "tf plan validate all": │   on .terraform/modules/data_fusion/modules/instance/versions.tf line 18, in terraform:
Step #1 - "tf plan validate all": │   18:   required_version = "~> 0.12.6"
Step #1 - "tf plan validate all": │ 
Step #1 - "tf plan validate all": │ Module module.data_fusion.module.instance (from ./modules/instance) does
Step #1 - "tf plan validate all": │ not support Terraform version 1.0.6. To proceed, either choose another
Step #1 - "tf plan validate all": │ supported Terraform version or update this version constraint. Version
Step #1 - "tf plan validate all": │ constraints are normally set for good reason, so updating the constraint
Step #1 - "tf plan validate all": │ may lead to other errors or unexpected behavior.`

### Terraform Configuration

```hcl
module "data_fusion" {
  source  = "terraform-google-modules/data-fusion/google"
  version = "1.0.0"

  name    = "df-test-instance"
  project = "df-test-instance"
  region  = "us-central1"
}
```

### Terraform Version

```sh
1.0.6
```
